client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -124,6 +124,23 @@
    boolean replaceWithVersion(K key, V newValue, long version, int lifespanSeconds, int maxIdleTimeSeconds);
 
    /**
+    * A overloaded form of {@link #replaceWithVersion(Object, Object, long)}
+    * which takes in lifespan and maximum idle time parameters.
+    *
+    * @param key key to use
+    * @param newValue new value to be associated with the key
+    * @param version numeric version that should match the one in the server
+    *                for the operation to succeed
+    * @param lifespan lifespan of the entry
+    * @param lifespanTimeUnit {@link java.util.concurrent.TimeUnit} for lifespan
+    * @param maxIdle the maximum amount of time this key is allowed
+    *                           to be idle for before it is considered as expired
+    * @param maxIdleTimeUnit {@link java.util.concurrent.TimeUnit} for maxIdle
+    * @return true if the value was replaced
+    */
+   boolean replaceWithVersion(K key, V newValue, long version, int lifespan, TimeUnit lifespanTimeUnit, int maxIdle, TimeUnit maxIdleTimeUnit);
```

Comment:
I am just being picky here, but have we limited ourselves in the protocol to only supporting ints and not longs already?  I am just curious because embedded everything is a long.